src/components/UserDashboard/CavaticaProjects/CavaticaProvider.js

```diff
@@ -10,22 +10,24 @@ const CavaticaProvider = ({ children, onData }) => (
       const projects = await getCavaticaProjects();
       onData(projects);
 
-      const projectsWithData = await projects.map(async p => {
-        const members = await getMembers({ project: p.id });
-        const completedTasks = await getTasks({ project: p.id, type: 'COMPLETED' });
-        const failedTasks = await getTasks({ project: p.id, type: 'FAILED' });
-        const runningTasks = await getTasks({ project: p.id, type: 'RUNNING' });
+      Promise.all(
+        projects.map(async p => {
+          const [members, completedTasks, failedTasks, runningTasks] = await Promise.all([
+            getMembers({ project: p.id }),
+            getTasks({ project: p.id, type: 'COMPLETED' }),
+            getTasks({ project: p.id, type: 'FAILED' }),
+            getTasks({ project: p.id, type: 'RUNNING' }),
+          ]);
```

Comment:
@ciaranschutte for mutually independent async operations, this should help speed things up a little :)